glide-core/redis-rs/redis/src/cluster_async/connections_container.rs

```diff
@@ -275,20 +313,39 @@ where
         node: ClusterNode<Connection>,
     ) -> String {
         let address = address.into();
-        self.connection_map.insert(address.clone(), node);
+
+        // Increase the total number of connections
+        Telemetry::incr_total_connections(node.connections_count());
+
+        self.connection_map
+            .insert(address.clone(), node)
+            .map(|old_conn| {
+                // old_conn is dropped here -> decrease the count by the number of connections
+                // held by `old_conn`
+                // Increase the total number of connections
+                Telemetry::decr_total_connections(old_conn.connections_count());
+                old_conn
```

Comment:
why do you return it if the result of insert is being ignored?

Reply:
fixed